### PR TITLE
compatibility with jupyterlab >= 2.0

### PR DESCRIPTION
**Pull request by btel**
rebased and edited from #520 

Fixes issue #559 
Closes #520 

For release schedule see #602

### CODE REVIEW

**Comment by btel** [comment]
@fcollonval I rebased the PR #550 on master and fixed a couple of issues. All tests pass. Could you review?

**Comment by lresende** [comment]
@btel I have added a couple of minor changes, and at a first glance it's looking good. Let me spend a little more time with the applied changes.

**Comment by btel** [comment]
thanks @lresende. It looks good. Why do we need to force enable extensions here? Shouldn't it be applied in another PR to keep the single scope of this one?

**Comment by lresende** [comment]
@btel Agree with the assessment for the last commit. How do you want to handle that? Revert or Remove + push force? After your preference, I can update the branch...

**Comment by btel** [comment]
@lresende it's done! I just removed and force-pushed the branch.

File: setup.py [hunk 1]
```diff
@@ -71,14 +71,14 @@ def runPackLabextension():
...
-        'nbdime >= 1.1.0, < 2.0.0',
+        'nbdime >= 2.0.0',
         'pexpect'
     ],
     extras_require = {
         'test': [
             'pytest',
             'pytest-asyncio',
-            'jupyterlab~=1.1',
+            'jupyterlab>=2.0',
```
Comment (fcollonval) [inline setup.py:81]:
```suggestion
            'jupyterlab~=2.0',
```

File: setup.py [hunk 2]
```diff
@@ -71,14 +71,14 @@ def runPackLabextension():
     ],
     install_requires = [
         'notebook',
-        'nbdime >= 1.1.0, < 2.0.0',
+        'nbdime >= 2.0.0',
```
Comment (fcollonval) [inline setup.py:74]:
```suggestion
        'nbdime ~=2.0',
```

File: package.json [hunk 3]
```diff
@@ -49,30 +49,33 @@
...
+    "@lumino/polling": "^1.0.4",
+    "@lumino/widgets": "^1.11.1",
     "@material-ui/core": "^4.8.2",
     "@material-ui/icons": "^4.5.1",
-    "@phosphor/widgets": "^1.8.0",
     "diff-match-patch": "^1.0.4",
-    "nbdime": "~5.0.1",
-    "react": "~16.8.4",
-    "react-dom": "~16.8.4",
+    "nbdime": "~6.0.0",
```
Comment (fcollonval) [inline package.json:69]:
```suggestion
    "nbdime": "^6.0.0",
```

**Comment by btel** [comment]
@fcollonval thanks for the comments.  I will integrate them tonight. There is no problem with waiting for the fixes to be published before merging, but it's better not to wait to long so the master does not diverge too much from this PR.

**Comment by telamonian** [comment]
@ajbozarth Good icon fixes! But... posting the contents of a raw patch file (that I then have to cut and paste) isn't the friendliest way to share your changes. In the future, don't be shy about just pushing minor fixes to the PR branch

**Comment by ajbozarth** [comment]
@telamonian I tried just pushing this changes but it said I didn't have permissions for some reason

**Comment by telamonian** [comment]
@ajbozarth Strange. I would have thought being a member of juptyerlab would have been permissions enough.

Whelp, anyhow I just gave you write access for this repo specifically. Hopefully that means you'll be able to push to PRs in the future

**Comment by lresende** [comment]
@fcollonval I am not sure what you were requesting (or maybe someone bit me to it) as you have admin access to the repo. If you were looking to update this pr, just create a checkout of the original branch from @btel and push to it (which is what I and @telamonian did).

I have also created #602 for us to track the release process. I will have some time tomorrow afternoon to help with this.

**Comment by telamonian** [comment]
@lresende I gave @fcollonval admin rights

**Comment by btel** [comment]
Thanks all for comments/commit!

@fcollonval I merged your suggestions (it took me a while before relising that I can do it directly from github UI ;-)) 
@ajbozarth you suggestions has been already integrated, so we are good to go!

**Pull request merged**